### ISSUE TITLE
[Bug]: Matching books using Audible.com showing "No Result"

### ISSUE THREAD

**Issue by seannymurrs** [advplyr/audiobookshelf#3385]
### What happened?

Starting yesterday afternoon, I'm getting "No Results" every time I try to match a book using Audible.com. I've tried using the ASIN or the Title, and I get the same outcome. Below are the logs that I get when I try to match using the ASIN B0CKM5YQX8.

`2024-09-06 18:39:32.115
ERROR
[Audible] ASIN search error AggregateError
2024-09-06 18:39:33.249
ERROR
[Audible] query search error AggregateError
2024-09-06 18:39:33.250
INFO
[Audnexus] Searching for author "https://api.audnex.us/authors?name=b0ckm5yqx8"
2024-09-06 18:39:33.763
ERROR
[Audnexus] Author ASIN request failed for b0ckm5yqx8 AggregateError`

One thing that was pointed out to me when I asked about this in the Discord is the fact that the logs show that the ASIN is in all lowercase even though I enter it as all caps when searching. They pointed out that following the Audnexus url with the ASIN in all caps does provide a result, but the URL in the logs doesn't (presumably because it's in lowercase).

### What did you expect to happen?

I expected the book to be matched to the correct book via its ASIN.

### Steps to reproduce the issue

1. Search for book using ASIN.
2. See "No Results".

### Audiobookshelf version

v2.13.3

### How are you running audiobookshelf?

Docker

### What OS is your Audiobookshelf server hosted from?

macOS

### If the issue is being seen in the UI, what browsers are you seeing the problem on?

None

### Logs

```shell
2024-09-06 18:39:32.115
ERROR
[Audible] ASIN search error AggregateError
2024-09-06 18:39:33.249
ERROR
[Audible] query search error AggregateError
2024-09-06 18:39:33.250
INFO
[Audnexus] Searching for author "https://api.audnex.us/authors?name=b0ckm5yqx8"
2024-09-06 18:39:33.763
ERROR
[Audnexus] Author ASIN request failed for b0ckm5yqx8 AggregateError
```

### Additional Notes

I've tried the ASIN for other books as well as searching by Title and Author. I get "No Results" every time.

**Comment by hologram75** [comment]
I have the exact same issue (Docker Windows v2.13.3).

**Comment by advplyr** [comment]
Audnexus API website is down

**Comment by CoolJoe72** [comment]
Domain Audnex.us Expires on 2024-09-06. Hopefully it gets renewed.

**Comment by valdearg** [comment]
docker compose people can also use 

```
    extra_hosts:
     - "audnex.us:104.21.44.87"
     - "api.audnex.us:104.21.44.87"
```

**Comment by Abateman121** [comment]
> docker compose people can also use
> 
> ```
>     extra_hosts:
>      - "audnex.us:104.21.44.87"
>      - "api.audnex.us:104.21.44.87"
> ```

You are a hero good person!

**Comment by catduckgnaf** [comment]
And legend among men

**Comment by petenun** [comment]
Thank you. I added those entries to my hosts file (Windows) and it's working nicely. The question is, what happened to the audnex.us domain registration in DNS. It's just gone.

**Comment by stibbons1990** [comment]
Kubernetes people can use

```yaml
      hostAliases:
      - ip: "104.21.44.87"
        hostnames:
        - "api.audnex.us"
        - "audnex.us"
      containers:
        - image: ghcr.io/advplyr/audiobookshelf:latest
      ...
```

**Comment by sebinum** [comment]
If you're on Unraid:

1. Edit the container
2. Switch from **Basic View** to **Advanced View** (can be toggled on the top right)
3. Add the following into the **Extra Parameters** field and restart:

```
--add-host=audnex.us:104.21.44.87 --add-host=api.audnex.us:104.21.44.87
```

**Comment by xcy7e** [comment]
If you use the linux package version, do this:
```bash
sudo echo "104.21.44.87 api.audnex.us" >> /etc/hosts && sudo echo "104.21.44.87 audnex.us" >> /etc/hosts
```

or add it manually to `/etc/hosts`:
```bash
# ... 

104.21.44.87 api.audnex.us
104.21.44.87 audnex.us
```

**Comment by advplyr** [comment]
Audnexus is back up

**Comment by ZLoth** [comment]
Hopefully, this won't occur again next September. Per https://github.com/laxamentumtech/audnexus/issues/808: "Yes, the auto renew failed. I've manually fixed it. Should be back soon."

**Comment by seannymurrs** [comment]
I'm still seeing the same issue I originally was.

**Comment by MaxPowers1337** [comment]
Jeah, same probleme here, too.

For me it seems that the audnex api is not getting the correct values from audible itself.

ive created a [issue there.](https://github.com/djdembeck/Audnexus.bundle/issues/122)

**Comment by xcy7e** [comment]
> Jeah, same probleme here, too.
> 
> For me it seems that the audnex api is not getting the correct values from audible itself.
> 
> ive created an [issue there.](https://github.com/djdembeck/Audnexus.bundle/issues/122)

Hm, can't reproduce - works fine here, if I avoid the DNS by mapping the APIs IP directly to the domains using the `hosts` file.

Even direct search for `ASIN` works.

**Comment by MaxPowers1337** [comment]
> > Jeah, same probleme here, too.
> > For me it seems that the audnex api is not getting the correct values from audible itself.
> > ive created an [issue there.](https://github.com/djdembeck/Audnexus.bundle/issues/122)
> 
> Hm, can't reproduce - works fine here, if I avoid the DNS by mapping the APIs IP directly to the domains using the `hosts` file.
> 
> Even direct search for `ASIN` works.
> 
> Anyways this is no audiobookshelf-related issue and should not be reported here, hence this issue is already closed.

The url works for you?

https://api.audnex.us/books/B0DFHJFZQH?region=de

**Comment by xcy7e** [comment]
Everything within audiobookshelf works for me, relating this context.

**Comment by MaxPowers1337** [comment]
Could you kindly check if the above ASIN works for you, too? Audiobookshelf is not able to match this item and i keep getting this error : 

```
    data: {
      statusCode: 500,
      error: 'Internal Server Error',
      message: "Required key 'asin' does not exist in Audible API response for ASIN B0DFHJFZQH"
    }
```

I strongly suspect that audnexus has a problem with this specific item and that the error is not on audiobookshelf's side.

**Comment by petenun** [comment]
I just looked in DNS and it looks like audnex.us has a split domain, pointing to multiple IP addresses. Since this works for some people but not others, it seems feasible that one of the IP addresses could be broken. The 104.21.44.87 address seems to work for me, so you may want to keep using the modified host file until a fix is in place.

NSLOOKUP

> audnex.us
Server:  dns.google
Address:  8.8.8.8

Non-authoritative answer:
Name:    audnex.us
Addresses:  2606:4700:3030::ac43:c637
          2606:4700:3037::6815:2c57
          104.21.44.87
          172.67.198.55

**Comment by wesf99** [comment]
I have been getting this error on my synology nas docker installation for the past few days. 

I have added:
104.21.44.87 api.audnex.us
104.21.44.87 audnex.us
to my /etc/hosts

This does not seem to have helped me.

**Comment by nichwall** [comment]
> I have been getting this error on my synology nas docker installation for the past few days. 
> 
> I have added:
> 104.21.44.87 api.audnex.us
> 104.21.44.87 audnex.us
> to my /etc/hosts

Are you getting the same error as mentioned in the original message, and have you checked that you get a response when manually querying api.audnex.us?

**Comment by wesf99** [comment]
Thank you for your reply. Looking more closely, I am getting a different error: 

> Error: getaddrinfo EAI_AGAIN api.audnex.us


I was able to get a response from api.audnex.us outside of ABS.

I realized I had recently installed another docker compose app and had to add a new firewall rule on my Synology to allow the new docker IP address.  I added a new firewall rule to allow the ABS IP network and it fixed the problem. I'm not sure what happened as ABS had been working for me for over 12 to 18 months.

**Comment by wesf99** [comment]
> > I have been getting this error on my synology nas docker installation for the past few days.
> > I have added:
> > 104.21.44.87 api.audnex.us
> > 104.21.44.87 audnex.us
> > to my /etc/hosts
> 
> Are you getting the same error as mentioned in the original message, and have you checked that you get a response when manually querying api.audnex.us outside of ABS?

Thank you for your reply. Looking more closely, I am getting a different error:

Error: getaddrinfo EAI_AGAIN api.audnex.us

I was able to get a response from api.audnex.us outside of ABS.

I realized I had recently installed another docker compose app and had to add a firewall rule on my Synology to allow the new docker IP address. I added a new firewall rule to allow the ABS IP network and it fixed the problem. I'm not sure what happened as ABS had been working for me for over 12 to 18 months without needing a rule!

Thanks again for getting me to look more closely.
